python/tvm/relay/to_onnx.py

```diff
@@ -0,0 +1,409 @@
+            'epsilon': float(attrs.get_str('epsilon')),
+            # 'spatial' : 1 #TODO - version based support
+            }
+
+
+class Dropout(OpConverter):
+    """ Operator converter for Dropout.
+    """
+    @classmethod
+    def convert_attributes(cls, attrs):
+        return {
+            'ratio': float(attrs.get_str('rate')),
+        }
+
+
+class AveragePool(MaxPool):
+    """ Operator converter for AveragePool.
+    """
+
+
+class Concat(OpConverter):
+    """ Operator converter for Concat.
+    """
+    @classmethod
+    def convert_attributes(cls, attrs):
+        return {
+            'axis': attrs.get_int("axis"),
+            }
+
+
+class BiasAdd(OpConverter):
+    """ Operator converter for BiasAdd.
+    """
+
+    @classmethod
+    def convert(cls, node, model_container, node_list):
+
+        input_node = node_list[node['inputs'][0][0]]
+        data_ndim = len(input_node['types'][0].shape)
+        axis = node['node'].attrs.get_int("axis")
+        if axis < 0:
+            axis = axis + data_ndim
+        num_newaxis = data_ndim - axis - 1
+        if num_newaxis:
+            output_name = 'inter{}'.format(node['output_names'][0])
+            transpose_node = onnx.helper.make_node('Unsqueeze',
+                                                   [node['input_names'][1]],
+                                                   [output_name],
+                                                   **{'axes': tuple(range(1, num_newaxis+1))})
+            model_container.add_nodes([transpose_node])
+        else:
+            output_name = node['input_names'][1]
+
+        inputs = [node['input_names'][0], output_name]
+        matmul_node = onnx.helper.make_node('Add', inputs, node['output_names'])
+        model_container.add_nodes([matmul_node])
+
+
+relay_to_onnx_op_mapping = {
+    'reshape': Reshape,
+    'conv2d': Conv,
+    'add': rename('Add'),
+    'relu': rename('Relu'),
+    'transpose': Transpose,
+    'dense': MatMul,
+    'max_pool2d': MaxPool,
+    'batch_flatten': Flatten,
+    'multiply': rename('Mul'),
+    'bias_add': BiasAdd,
+    'batch_norm': BatchNormalization,
+    'global_avg_pool2d': rename('GlobalAveragePool'),
+    'concatenate': Concat,
+    'dropout': Dropout,
+    'avg_pool2d': AveragePool,
+    'divide': rename('Div')
+}
+
+
+class ModelContainer(object):
+    """ A container class to hold  different attributes of ONNX model graph
+    """
+
+    def __init__(self, name):
+        self._name = name
+        self._inputs = []
+        self._outputs = []
+        self._nodes = []
+        self._initializers = []
+
+    def add_inputs(self, inputs):
+        self._inputs.extend(inputs)
+
+    def add_outputs(self, outputs):
+        self._outputs.extend(outputs)
+
+    def add_nodes(self, nodes):
+        self._nodes.extend(nodes)
+
+    def add_initializers(self, initializers):
+        self._initializers.extend(initializers)
+
+    def make_model(self):
+        graph_def = onnx.helper.make_graph(
```

Comment:
how about `onnx_graph`? since graph_def is specific name used in TF

Reply:
changed